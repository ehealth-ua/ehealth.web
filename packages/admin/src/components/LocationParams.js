import React from "react";
import { Location } from "@reach/router";
import { parseSearchParams, stringifySearchParams } from "@ehealth/utils";

const LocationParams = ({ children, render = children }) => (
  <Location
    children={({ location, navigate }) =>
      render({
        locationParams: parseSearchParams(location.search),
        setLocationParams: params => {
          const query = stringifySearchParams(params);
          if (location.search === `?${query}`) return;
          navigate(`${location.pathname}?${query}`);
        }
      })
    }
  />
);

export default LocationParams;
