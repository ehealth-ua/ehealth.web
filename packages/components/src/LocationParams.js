import React from "react";
import { Location } from "@reach/router";
import { parseSearchParams, stringifySearchParams } from "@ehealth/utils";
import isEqual from "lodash/isEqual";

const LocationParams = ({ children, render = children }) => (
  <Location
    children={({ location, navigate }) =>
      render({
        locationParams: parseSearchParams(location.search),
        setLocationParams: params => {
          const parsedParams = parseSearchParams(location.search);
          if (Object.entries(params).length && !isEqual(parsedParams, params)) {
            const query = stringifySearchParams(params);
            navigate(`${location.pathname}?${query}`);
          }
        }
      })
    }
  />
);

export default LocationParams;
