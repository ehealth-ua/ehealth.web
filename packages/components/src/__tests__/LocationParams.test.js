import React from "react";
import renderer from "react-test-renderer";
import LocationParams from "../LocationParams";

describe("LocationParams", () => {
  it("should render correctly", () => {
    const component = renderer.create(
      <LocationParams>{() => <div />}</LocationParams>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it("should return object in render prop", () => {
    const render = jest.fn(() => <div />);
    const component = renderer.create(
      <LocationParams>{render}</LocationParams>
    );
    expect(render).toHaveBeenCalledTimes(1);
    expect(render).toMatchSnapshot();
  });
});
