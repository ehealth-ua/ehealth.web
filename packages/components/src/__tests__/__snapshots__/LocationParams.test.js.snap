// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationParams should render correctly 1`] = `<div />`;

exports[`LocationParams should return object in render prop 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "locationParams": Object {},
        "setLocationParams": [Function],
      },
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": <div />,
    },
  ],
}
`;
